ethereum/core/src/main/java/org/hyperledger/besu/ethereum/MainnetBlockValidator.java

```diff
@@ -201,7 +218,11 @@ private void handleAndLogImportFailure(
         LOG.info("Invalid block {}", invalidBlock.toLogString());
       }
     }
-    badBlockManager.addBadBlock(invalidBlock, result.causedBy());
+    if (shouldRecordBadBlock) {
+      badBlockManager.addBadBlock(invalidBlock, result.causedBy());
+    } else {
+      LOG.debug("Invalid block not added to badBlockManager {}", invalidBlock.toLogString());
```

Comment:
Good to include the result.causedBy() in the log as well

Reply:
Agree, if it's not already logged

Reply:
the causeBy is already logged if present on line 207